### PR TITLE
TLS 1.3: Update documentation for the coming release and misc

### LINKED ISSUE #6379: Update TLS 1.3 docs for 3.3 release
Ensure TLS 1.3 documentation is accurate / up-to-date

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
## Status
READY

## Requires Backporting
No

Fix #6379

### CODE REVIEW

**Review by yuhaoth — DISMISSED** [review]
LGTM

File: ChangeLog.d/tls13-misc.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Features
+   * Mbed TLS supports TLS 1.3 key establishment via pre-shared keys,
+     pre-shared keys provisioned externally or via the ticket mechanism
+     (session resumption).
+     The MBEDTLS_SSL_SESSION_TICKETS configuration option controls the support
+     of the ticket mechanism.
+     MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_xxx_ENABLED configuration options
+     have been introduced to control the support of the three possible
```
Comment (xkqian) [inline ChangeLog.d/tls13-misc.txt:8]:
```suggestion
     have been introduced to control the support for the three possible
```

File: docs/architecture/tls13-support.md [hunk 2]
```diff
@@ -54,8 +57,8 @@ Support description
   | server_certificate_type      | no      |
   | padding                      | no      |
   | key_share                    | YES     |
-  | pre_shared_key               | no      |
-  | psk_key_exchange_modes       | no      |
+  | pre_shared_key               | YES     |
+  | psk_key_exchange_modes       | YES     |
   | early_data                   | no      |
   | cookie                       | no      |
```
Comment (xkqian) [inline docs/architecture/tls13-support.md:63]: Do we support cookie?  or only part of them?

**Review by xkqian — COMMENTED** [review]
Seems we chang the file tls13_compat.sh, should we change the python script tests/scripts/generate_tls13_compat_tests.py?

File: docs/architecture/tls13-support.md (hunk 2 above)
Comment (ronald-cron-arm) [reply docs/architecture/tls13-support.md:63]: Only partially I think and at least we do not have any test for it.

File: ChangeLog.d/tls13-misc.txt (hunk 1 above)
Comment (ronald-cron-arm) [reply ChangeLog.d/tls13-misc.txt:8]: Indeed, it seems I have difficulties to assimilate this.

**Review by xkqian — COMMENTED** [review]
LGTM

**Review by xkqian — APPROVED** [review]

**Comment by ronald-cron-arm** [comment]
@yuhaoth please have a look to the last version.

**Review by yuhaoth — APPROVED** [review]
LGTM

**Pull request merged**